packages/backend/src/extension.spec.ts

```diff
@@ -19,23 +19,17 @@
 import { beforeEach, expect, test, vi } from 'vitest';
 import type { ExtensionContext } from '@podman-desktop/api';
 import { activate, deactivate } from './extension';
+import { Studio } from './studio';
 
-const mocks = vi.hoisted(() => ({
-  studioActivateMock: vi.fn(),
-  studioDeactivateMock: vi.fn(),
-  studioConstructor: vi.fn(),
-}));
-
-vi.mock('./studio', () => ({
-  Studio: mocks.studioConstructor,
-}));
+vi.mock('./studio', () => {
+  const Studio = vi.fn();
+  Studio.prototype.activate = vi.fn();
+  Studio.prototype.deactivate = vi.fn();
+  return { Studio };
+});
```

Comment:
I would argue we should avoid using the `Class#prototype` maybe something like the following is an alternative

```ts
import { Studio } from './studio';

const STUDIO_MOCK: Studio = {
  activate: vi.fn(),
  deactivate: vi.fn(),
} as unknown as Studio;

vi.mock('./studio', () => ({
  Studio: vi.fn(),
}));

beforeEach(() => {
  vi.resetAllMock();
  vi.mocked(Studio).mockReturnValue(STUDIO_MOCK);
});

test('ensure activate', () => {
  // ....
  
  expect(Studio).toBeCalledTimes(1);
  expect(STUDIO_MOCK.activate).toHaveBeenCalled();
});
```

Reply:
With their solution, it is annoying to do `expect().haveBeenCalledWith(??)` because you have the specify the `this`. 

We would need to use `expect(Studio.prototype.activate).toHaveBeenCalledWith(expect.anything(), ...arguments)` :thinking: .

Moreover this prevent us from mocking multiple value. I cannot do `vi.mocked(studio).mockReturnOnce()` then another..

I understand that them having this in their documentation is a strong argument, but I am not a big fan of playing with Class#prototype :(